### PR TITLE
[symbology] Fix setSize(size) not working for ellipse markers

### PR DESCRIPTION
**Pull request by nirvn**
## Description
This has been frustrating me one time too many: until now, the ellipse marker's setSize function had an implementation gap and was not having any impact on the marker's width/height. This lead to users being unable to use a marker symbol's overall size spinbox to modify the size of an ellipse marker symbol layer.

This PR fixes that:
peek7CYH1Z

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/core/symbology/qgsellipsesymbollayer.cpp [hunk 1]
```diff
@@ -593,6 +593,32 @@ void QgsEllipseSymbolLayer::preparePath( const QString &symbolName, QgsSymbolRen
   }
 }
 
+void QgsEllipseSymbolLayer::setSize( double size )
+{
+  double ratio = size / mSize;
+  mSize = size;
```
Comment (nyalldawson) [inline src/core/symbology/qgsellipsesymbollayer.cpp]:
```suggestion
  QgsMarkerSymbolLayer::setSize( size );
```

**Comment by nyalldawson** [comment]
Nice! Care to add some tests for this? Ellipse marker will thank you!

**Comment by nirvn** [comment]
@nyalldawson , tests added just as you were hitting the comments button :smile:

File: src/core/symbology/qgsellipsesymbollayer.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/symbology/qgsellipsesymbollayer.cpp]: @nyalldawson , good point.

**Pull request merged**